jujugui/static/gui/src/app/components/inspector/unit-list/unit-list.js

```diff
@@ -125,14 +125,21 @@ YUI.add('unit-list', function() {
       @param {String} action The action to apply to the units.
     */
     _handleUpdateUnits: function(action) {
-      var units = [];
-      var refs = this.refs;
-      var envResolved = this.props.envResolved;
+      let unitNames = [];
+      const units = this.props.units;
+      const refs = this.refs;
+      const envResolved = this.props.envResolved;
       Object.keys(refs).forEach(function (ref) {
-        var isInstance = ref.split('-')[0] === 'CheckListItem';
+        let isInstance = ref.split('-')[0] === 'CheckListItem';
         if (isInstance && refs[ref].state.checked) {
-          var unitName = ref.slice(ref.indexOf('-') + 1);
-          units.push(unitName);
+          let unitName = ref.slice(ref.indexOf('-') + 1);
+          unitNames.push(unitName);
+          const unit = units.filter((u) => {
+            return u.id == unitName;
+          })[0];
+          if (!unit.agent_status) {
```

Comment:
I'm assuming that the rational behind not creating a db method for this is because we don't have access to the unit list model. So can you add a code comment as to why this is necessary as it's super cryptic.